bundlepath.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package charmrepo
+
+import (
+	"os"
+	"path/filepath"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+)
+
+// BundlePath is used to access a bundle at a given path.
+type BundlePath interface {
+	// Bundle returns the bundle represented by this path.
+	Bundle() (charm.Bundle, *charm.URL)
+}
+
+// bundlePath represents a path (eg local directory) containing a
+// single bundle.
+type bundlePath struct {
+	Name       string
+	BundleInfo charm.Bundle
+}
+
+var _ BundlePath = (*bundlePath)(nil)
+
+// NewBundlePath creates and returns an instance used to
+// open a bundle located at the given path.
+func NewBundlePath(path string) (BundlePath, error) {
```

Comment:
I was thinking we may want to add other methods to the struct later. Juju core doesn't yet deploy local bundles directly so it's hard to know what's required. I thought it would be more consistent to be the same as for charm and allow room for extension later.